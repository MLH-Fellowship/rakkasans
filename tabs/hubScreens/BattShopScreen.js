import React, { useEffect, useState } from "react";
import { View, Text, StyleSheet, Image } from "react-native";
import { Table, Rows } from "react-native-table-component";
import { useNavigation } from "@react-navigation/native";

import Colors from "../../constants/Colors";
import RectButton from "../../components/RectButton";
import Dimensions from "../../constants/Dimensions";

export default function BattShopScreen ({ route }) {
  const [batt, setBatt] = useState({});
  const { name } = route.params;
  const {shop_number=false, hours=false, location=false} = batt
  useEffect(() => {
    const getBatt = async () => {
      try {
        // TODO Change to localhost for non Mike development
        const response = await fetch(`http://localhost:3000/battalion/${name}`);
        const json = await response.json();
        setBatt(json[0]);
      } catch (error) {
        console.log(error);
      }
    };

    getBatt();
  }, []);

  const tableData = [
    ["Shop #:", `${shop_number ? shop_number : '...loading'}`],
    ["Location:", `${location ? location : '...loading'}`],
    ["Hours:", `${hours ? hours: '...loading'}`]
  ];
  const titleWidth = Dimensions.window.width * 0.27;
  const contentWidth = Dimensions.window.width * 0.65;

  // const navigation = useNavigation();
  // const { name } = route.params;
  // const { image } = route.params;
  return (
    <>
    <View style={styles.container}>
      <View style={styles.buttonContainer}>
        {/* <Text style={styles.title}> {name} </Text>
        <Image source={image} style={styles.titlePic} /> */}
        <Table borderStyle={{ borderWidth: 1, borderColor: Colors.lightGray }}>
          <Rows
            data={tableData}
            textStyle={[styles.tableText, Dimensions.font]}
            widthArr={[titleWidth, contentWidth]}
          />
        </Table>
        <RectButton text="Contact Info/Details" />
        <RectButton text="Security Clearance" />
        <RectButton text="Regulations" />
        <RectButton text="Function" />
      </View>
    </View>
    </>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    // justifyContent: "center",
    alignItems: "center",
    backgroundColor: Colors.white
  },
  title: {
    fontSize: 36,
    color: Colors.darkGray,
    fontFamily: "andale-mono",
    marginBottom: 10
  },
  titlePic: {
    width: 130,
    height: 130,
    top: "-5%",
    marginTop: 20
  },
  titleContainer: {
    alignItems: "center",
    marginTop: 80
  },
  buttonContainer: {
    marginTop: 20,
    width: "90%"
  },
  buttonView: {
    marginTop: 35,
    flexDirection: "row",
    alignItems: "center",
    justifyContent: "space-evenly",
    paddingBottom: 15,
    width: "100%"
  },
  tableText: {
    fontSize: 16,
    margin: 6
  }
});
