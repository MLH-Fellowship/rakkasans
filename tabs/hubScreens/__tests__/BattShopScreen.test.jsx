import React from 'react';
import renderer from 'react-test-renderer';

import BattShopScreen from '../BattShopScreen';

jest.useFakeTimers();

it('renders correctly', () => {
  const tree = renderer.create(<BattShopScreen />).toJSON();
  expect(tree).toMatchSnapshot();
});
