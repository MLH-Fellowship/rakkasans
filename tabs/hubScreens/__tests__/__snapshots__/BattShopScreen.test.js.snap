// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#ffffff",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "marginTop": 20,
        "width": "90%",
      }
    }
  >
    <View
      style={
        Array [
          undefined,
          Object {
            "borderBottomWidth": 1,
            "borderColor": "#efefef",
            "borderLeftWidth": 1,
          },
        ]
      }
    >
      <View
        style={
          Array [
            0,
            Object {
              "width": 690,
            },
          ]
        }
      >
        <View
          style={
            Array [
              undefined,
              Object {
                "width": 690,
              },
              Object {
                "flexDirection": "row",
                "overflow": "hidden",
              },
              undefined,
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "borderColor": "#efefef",
                  "borderRightWidth": 1,
                  "borderTopWidth": 1,
                },
                Object {
                  "justifyContent": "center",
                },
                Object {
                  "width": 202.5,
                },
                undefined,
                undefined,
                false,
                undefined,
              ]
            }
          >
            <Text
              style={
                Array [
                  Array [
                    Object {
                      "fontSize": 16,
                      "margin": 6,
                    },
                    Object {
                      "fontFamily": "andale-mono",
                    },
                  ],
                  Object {
                    "backgroundColor": "transparent",
                  },
                ]
              }
            >
              Shop #:
            </Text>
          </View>
          <View
            style={
              Array [
                Object {
                  "borderColor": "#efefef",
                  "borderRightWidth": 1,
                  "borderTopWidth": 1,
                },
                Object {
                  "justifyContent": "center",
                },
                Object {
                  "width": 487.5,
                },
                undefined,
                undefined,
                false,
                undefined,
              ]
            }
          >
            <Text
              style={
                Array [
                  Array [
                    Object {
                      "fontSize": 16,
                      "margin": 6,
                    },
                    Object {
                      "fontFamily": "andale-mono",
                    },
                  ],
                  Object {
                    "backgroundColor": "transparent",
                  },
                ]
              }
            >
              xxx-xxx-xxxx
            </Text>
          </View>
        </View>
        <View
          style={
            Array [
              undefined,
              Object {
                "width": 690,
              },
              Object {
                "flexDirection": "row",
                "overflow": "hidden",
              },
              undefined,
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "borderColor": "#efefef",
                  "borderRightWidth": 1,
                  "borderTopWidth": 1,
                },
                Object {
                  "justifyContent": "center",
                },
                Object {
                  "width": 202.5,
                },
                undefined,
                undefined,
                false,
                undefined,
              ]
            }
          >
            <Text
              style={
                Array [
                  Array [
                    Object {
                      "fontSize": 16,
                      "margin": 6,
                    },
                    Object {
                      "fontFamily": "andale-mono",
                    },
                  ],
                  Object {
                    "backgroundColor": "transparent",
                  },
                ]
              }
            >
              Location:
            </Text>
          </View>
          <View
            style={
              Array [
                Object {
                  "borderColor": "#efefef",
                  "borderRightWidth": 1,
                  "borderTopWidth": 1,
                },
                Object {
                  "justifyContent": "center",
                },
                Object {
                  "width": 487.5,
                },
                undefined,
                undefined,
                false,
                undefined,
              ]
            }
          >
            <Text
              style={
                Array [
                  Array [
                    Object {
                      "fontSize": 16,
                      "margin": 6,
                    },
                    Object {
                      "fontFamily": "andale-mono",
                    },
                  ],
                  Object {
                    "backgroundColor": "transparent",
                  },
                ]
              }
            >
              Example address which will be dynamically replaced
            </Text>
          </View>
        </View>
        <View
          style={
            Array [
              undefined,
              Object {
                "width": 690,
              },
              Object {
                "flexDirection": "row",
                "overflow": "hidden",
              },
              undefined,
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "borderColor": "#efefef",
                  "borderRightWidth": 1,
                  "borderTopWidth": 1,
                },
                Object {
                  "justifyContent": "center",
                },
                Object {
                  "width": 202.5,
                },
                undefined,
                undefined,
                false,
                undefined,
              ]
            }
          >
            <Text
              style={
                Array [
                  Array [
                    Object {
                      "fontSize": 16,
                      "margin": 6,
                    },
                    Object {
                      "fontFamily": "andale-mono",
                    },
                  ],
                  Object {
                    "backgroundColor": "transparent",
                  },
                ]
              }
            >
              Hours:
            </Text>
          </View>
          <View
            style={
              Array [
                Object {
                  "borderColor": "#efefef",
                  "borderRightWidth": 1,
                  "borderTopWidth": 1,
                },
                Object {
                  "justifyContent": "center",
                },
                Object {
                  "width": 487.5,
                },
                undefined,
                undefined,
                false,
                undefined,
              ]
            }
          >
            <Text
              style={
                Array [
                  Array [
                    Object {
                      "fontSize": 16,
                      "margin": 6,
                    },
                    Object {
                      "fontFamily": "andale-mono",
                    },
                  ],
                  Object {
                    "backgroundColor": "transparent",
                  },
                ]
              }
            >
              This is the hour schedule
            </Text>
          </View>
        </View>
      </View>
    </View>
    <View
      accessible={true}
      focusable={false}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#020a41",
          "borderRadius": 5,
          "flexDirection": "row",
          "justifyContent": "center",
          "marginVertical": 10,
          "opacity": 1,
          "padding": 10,
          "width": "100%",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "color": "#ffffff",
              "fontSize": 18,
              "marginHorizontal": 30,
              "marginTop": 7,
            },
            Object {
              "fontFamily": "andale-mono",
            },
          ]
        }
      >
        Contact Info/Details
      </Text>
    </View>
    <View
      accessible={true}
      focusable={false}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#020a41",
          "borderRadius": 5,
          "flexDirection": "row",
          "justifyContent": "center",
          "marginVertical": 10,
          "opacity": 1,
          "padding": 10,
          "width": "100%",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "color": "#ffffff",
              "fontSize": 18,
              "marginHorizontal": 30,
              "marginTop": 7,
            },
            Object {
              "fontFamily": "andale-mono",
            },
          ]
        }
      >
        Security Clearance
      </Text>
    </View>
    <View
      accessible={true}
      focusable={false}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#020a41",
          "borderRadius": 5,
          "flexDirection": "row",
          "justifyContent": "center",
          "marginVertical": 10,
          "opacity": 1,
          "padding": 10,
          "width": "100%",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "color": "#ffffff",
              "fontSize": 18,
              "marginHorizontal": 30,
              "marginTop": 7,
            },
            Object {
              "fontFamily": "andale-mono",
            },
          ]
        }
      >
        Regulations
      </Text>
    </View>
    <View
      accessible={true}
      focusable={false}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#020a41",
          "borderRadius": 5,
          "flexDirection": "row",
          "justifyContent": "center",
          "marginVertical": 10,
          "opacity": 1,
          "padding": 10,
          "width": "100%",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "color": "#ffffff",
              "fontSize": 18,
              "marginHorizontal": 30,
              "marginTop": 7,
            },
            Object {
              "fontFamily": "andale-mono",
            },
          ]
        }
      >
        Function
      </Text>
    </View>
  </View>
</View>
`;
